controllers/sharhSearch.controller.js

```diff
@@ -180,7 +166,13 @@ class SharhSearchController {
       .filter((id) => id !== undefined && id !== '0');
 
     if (sharhIds.length === 0) {
-      throw new AppError('No sharh found for the given search', 404);
+      return sendSuccess(res, 200, [], {
+        length: 0,
+        page: req.query.page,
+        removeHTML: req.isRemoveHTML,
+        specialist: req.isForSpecialist,
+        isCached: false,
+      });
```

Comment:
This will not be cached, so you can check if `sharhIds.length !== 0` then call the promise all
otherwise continue as normal without throw any error